src/Files.App/Filesystem/Security/AccessControlEntryAdvanced.cs

```diff
@@ -0,0 +1,454 @@
+		public InheritanceFlags InheritanceFlags
+		{
+			get => _inheritanceFlags;
+			set
+			{
+				if (SetProperty(ref _inheritanceFlags, value))
+					OnPropertyChanged(nameof(InheritanceFlagsHumanized));
+			}
+		}
+
+		private PropagationFlags _propagationFlags;
+		public PropagationFlags PropagationFlags
+		{
+			get => _propagationFlags;
+			set
+			{
+				if (SetProperty(ref _propagationFlags, value))
+					OnPropertyChanged(nameof(InheritanceFlagsHumanized));
+			}
+		}
+
+		private AccessMaskFlags _accessMaskFlags;
+		public AccessMaskFlags AccessMaskFlags
+		{
+			get => _accessMaskFlags;
+			set
+			{
+				if (SetProperty(ref _accessMaskFlags, value))
+					OnPropertyChanged(nameof(AccessMaskFlagsHumanized));
+			}
+		}
+
+		public string AccessMaskFlagsHumanized
+			=> string.Join(", ", HumanizeAccessMask());
+
+		public string IsInheritedHumanized
+			=> IsInherited ? "Yes".GetLocalizedResource() : "No".GetLocalizedResource();
+
+		public string InheritanceFlagsHumanized
+			=> string.Join(", ", HumanizeInheritance());
+
+		private bool _isSelected;
+		public bool IsSelected
+		{
+			get => _isSelected;
+			set
+			{
+				if (SetProperty(ref _isSelected, value))
+				{
+					if (!value)
+						AreAdvancedPermissionsShown = false;
+
+					OnPropertyChanged(nameof(IsEditEnabled));
+				}
+			}
+		}
+
+		private bool _areAdvancedPermissionsShown;
+		public bool AreAdvancedPermissionsShown
+		{
+			get => _areAdvancedPermissionsShown;
+			set
+			{
+				// Reconstruct list
+				if (SetProperty(ref _areAdvancedPermissionsShown, value))
+					AccessMaskItemList = GetAllAccessMaskList();
+			}
+		}
+
+		public bool IsEditEnabled
+			=> IsSelected && !IsInherited;
+
+		public bool IsFolder { get; }
+
+		private List<AccessMaskItem> _accessMaskItemList;
+		public List<AccessMaskItem> AccessMaskItemList
+		{
+			get => _accessMaskItemList;
+			set => SetProperty(ref _accessMaskItemList, value);
+		}
+
+		public bool WriteAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.Write);
+		public bool ReadAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.Read);
+		public bool ListDirectoryAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.ListDirectory);
+		public bool ReadAndExecuteAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.ReadAndExecute);
+		public bool ModifyAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.Modify);
+		public bool FullControlAccess => AccessMaskFlags.HasFlag(AccessMaskFlags.FullControl);
+		public bool SpecialAccess
+			=> (AccessMaskFlags &
+					~AccessMaskFlags.Synchronize &
+					(FullControlAccess ? ~AccessMaskFlags.FullControl : AccessMaskFlags.FullControl) &
+					(ModifyAccess ? ~AccessMaskFlags.Modify : AccessMaskFlags.FullControl) &
+					(ReadAndExecuteAccess ? ~AccessMaskFlags.ReadAndExecute : AccessMaskFlags.FullControl) &
+					(ReadAccess ? ~AccessMaskFlags.Read : AccessMaskFlags.FullControl) &
+					(WriteAccess ? ~AccessMaskFlags.Write : AccessMaskFlags.FullControl)) != 0;
```

Comment:
Indent looks strange here. Why are lines 164 and 165 splitted?

Reply:
Removed one tab each lines.